From: Ahmad Fatoum <a.fatoum@pengutronix.de>
Date: Tue, 31 May 2022 10:03:04 +0200
Subject: [PATCH] refactor(mmc): replace magic value with new
 PART_CFG_BOOT_PARTITION_NO_ACCESS

Disabling access to the boot partition reverts the MMC to read from the
user area. Add a macro to make this clearer.

Origin: https://review.trustedfirmware.org/c/TF-A/trusted-firmware-a/+/15337
Suggested-by: Manish V Badarkhe <Manish.Badarkhe@arm.com>
Signed-off-by: Ahmad Fatoum <a.fatoum@pengutronix.de>
Change-Id: I34a5a987980bb4690d08d255f465b11a4697ed5a
---
 drivers/mmc/mmc.c     | 2 +-
 include/drivers/mmc.h | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/mmc/mmc.c b/drivers/mmc/mmc.c
index 6b8456d1f4b9..39085395537f 100644
--- a/drivers/mmc/mmc.c
+++ b/drivers/mmc/mmc.c
@@ -778,7 +778,7 @@ int mmc_part_switch_user(void)
 {
 	int ret;
 
-	ret = mmc_part_switch(0);
+	ret = mmc_part_switch(PART_CFG_BOOT_PARTITION_NO_ACCESS);
 	if (ret < 0) {
 		ERROR("Failed to switch to user partition, %d\n", ret);
 	}
diff --git a/include/drivers/mmc.h b/include/drivers/mmc.h
index 2384b679f026..d18c5ffbdb6d 100644
--- a/include/drivers/mmc.h
+++ b/include/drivers/mmc.h
@@ -66,6 +66,7 @@
 #define EXT_CSD_PART_CONFIG_ACC_MASK	GENMASK(2, 0)
 #define PART_CFG_BOOT_PARTITION1_ENABLE	(U(1) << 3)
 #define PART_CFG_BOOT_PARTITION1_ACCESS (U(1) << 0)
+#define PART_CFG_BOOT_PARTITION_NO_ACCESS U(0)
 #define PART_CFG_BOOT_PART_EN_MASK		GENMASK(5, 3)
 #define PART_CFG_BOOT_PART_EN_SHIFT		3
 #define PART_CFG_CURRENT_BOOT_PARTITION(x)	(((x) & PART_CFG_BOOT_PART_EN_MASK) >> \
